test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/ListPackageTests.cs

```diff
@@ -156,6 +163,123 @@ public void BasicListPackage_OutputFormat_BadInput_Parsing_Fails(string outputFo
                 });
         }
 
+        [PlatformFact(Platform.Windows)]
+        public async Task ListPackage_WithPrivateHttpSourceCredentialServiceIsInvokedAsNeeded_Succeeds()
+        {
+            // Arrange
+            using var pathContext = new SimpleTestPathContext();
+
+            var packageA100 = new SimpleTestPackageContext("A", "1.0.0");
+            var packageB100 = new SimpleTestPackageContext("B", "1.0.0");
+
+            await SimpleTestPackageUtility.CreatePackagesAsync(
+                    pathContext.PackageSource,
+                    packageA100,
+                    packageB100);
+
+            var projectA = SimpleTestProjectContext.CreateNETCore("ProjectA", pathContext.SolutionRoot, "net6.0");
+            var projectB = SimpleTestProjectContext.CreateNETCore("ProjectB", pathContext.SolutionRoot, "net6.0");
+
+            projectA.AddPackageToAllFrameworks(packageA100);
+            projectB.AddPackageToAllFrameworks(packageB100);
+
+            var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+            solution.Projects.Add(projectA);
+            solution.Projects.Add(projectB);
+            solution.Create(pathContext.SolutionRoot);
+
+            SimpleTestSettingsContext.RemoveSource(pathContext.Settings.XML, "source");
```

Comment:
Not really a feedback on your code, since you're just using the APIs (I probably have to in the past but did not notice, hopefully I did not add them :D) 
These methods are a bit unusual. 
There's statics to operate on what's *always* the object being represented by the context :D

Reply:
In this test scenario, I removed the local folder source because I wanted the requests to be directed to the file-based mock server and credential service.